packages/amplify-cli-core/API.md

```diff
@@ -179,6 +179,42 @@ export enum AmplifyFrontend {
     javascript = "javascript"
 }
 
+// @public (undocumented)
+export type AmplifyNodeJsDetectorProps = {
+    projectRoot: string;
+    dependencyToSearch: string;
+    dependencyVersion: string;
+    packageManager: PackageManager;
+};
+
+// @public (undocumented)
+export class AmplifyNodePkgDetector {
+    constructor(amplifyDetectorProps: AmplifyNodeJsDetectorProps);
+    // (undocumented)
+    dependencyToSearch: string;
+    // (undocumented)
+    dependencyVersion: string;
+    // Warning: (ae-forgotten-export) The symbol "YarnLockDependencyType" needs to be exported by the entry point index.d.ts
+    // Warning: (ae-forgotten-export) The symbol "PackageLockDep" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
+    getDependentPackage(packageName: string): YarnLockDependencyType | PackageLockDep | undefined;
```

Comment:
like I mentioned in the other comment, package manager detection can be made internal given that `AmplifyNodePkgDetector` knows projectRoot path. When we do that we should change return type of `getDependentPackage` (or whatever comes out from next iteration) to something that just carries detected dependency version and name, i.e. information about package manager should not leak through this abstraction.